File: Assignment24/program5.c
/////////////////////////////////////////////////////////////////// 
//
// File name : program5.c
// Description : Accept string and display in reverse order
// Author : Adinath Santosh Pawar
// Date : 6/6/25
// 
///////////////////////////////////////////////////////////////////

#include<stdio.h>

void Reverse(char* str)
{
    int iLength = 0;
    if(NULL == str)
    {
        return;
    }

    while (*str != '\0')
    { 
        str++;
        iLength++;
    }

    str--;

    for (int i = iLength; i > 0; i--)
    {
        printf("%c",*str);
        str--;
    }
    

}

int main()
{
    char arr[20];

    printf("enter string \n");
    scanf("%[^'\n]s",arr);

    Reverse(arr);
   
    return 0;
}

File: Assignment45/program5.java
import java.util.*;

/////////////////////////////////////////////////////////////////// 
//
// File name : program5.java
// Description : Check is matrix is sparse matrix
// Author : Adinath Santosh Pawar
// Date : 2/8/25
// 
///////////////////////////////////////////////////////////////////

class Matrix
{
    private int Arr[][];

    public Matrix(int A, int B)
    {
        Arr = new int[A][B];
    }
        
    public void Accept()
    {
        System.out.println("Enter the elements of matrix ");

        Scanner sobj = new Scanner(System.in);

        int i = 0, j = 0;
        for(i = 0; i < Arr.length; i++)
        {
            for(j = 0; j < Arr[i].length; j++)
            {
                Arr[i][j] = sobj.nextInt();
            }
        }
    }

    public boolean ChkSparse(int iRow, int iCol) 
    {
        int CountZero = 0;
        int NonZero = 0;
        for (int i = 0; i < iRow; i++) 
        {
            for (int j = 0; j < iCol; j++) 
            {
                if(Arr[i][j] == 0)
                {
                    CountZero++;
                }
                else
                {
                    NonZero++;
                }
            }
        }

        return(CountZero > NonZero); 
    }


    public void Display()
    {
        System.out.println("The elements of matrix are ");

        int i = 0, j = 0;
        for(i = 0; i < Arr.length; i++)
        {
            for(j = 0; j < Arr[i].length; j++)
            {
                System.out.print(Arr[i][j] + " ");
            }
            System.out.println();
        }
    }

}
class program5
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);
        boolean bRet = false;

        System.out.println("Enter number of rows :");
        int iRow = sobj.nextInt();

        System.out.println("Enter number of columns :");
        int iCol = sobj.nextInt();

        Matrix mobj = new Matrix(iRow,iCol);

        mobj.Accept();
        bRet = mobj.ChkSparse(iRow,iCol);
        System.out.println(bRet);
    }
}